Samples/Samples/ViewModel/AppInfoViewModel.cs

```diff
@@ -15,9 +19,39 @@ public class AppInfoViewModel : BaseViewModel
 
         public Command OpenSettingsCommand { get; }
 
+        public IList<AppStateChange> History { get; } = new ObservableCollection<AppStateChange>();
+
         public AppInfoViewModel()
         {
             OpenSettingsCommand = new Command(() => AppInfo.OpenSettings());
+            AppInfo.AppStateChanged += UpdateState;
+        }
+
+        public override void OnDisappearing()
+        {
+            Task.Run(async () =>
```

Comment:
I think that this is maybe better to have at an app wide level perhaps. I think this sample is a bit odd. I don't like random Task.Run, even though I understand why it is there.

Reply:
Okay, would you prefer a static list in the AppInfoViewModel containing all state changes which is subscribed to on application start and never unsubscribed? Otherwise I dont see how we could hook this in the xamarin forms lifecycle, as OnSleep would be called prematurely.

Reply:
@jamesmontemagno I thought more about how app state could be sampled. The current app state while the ui is active is obvious, and on background there is no easy way to display the state. Would my proposal be acceptable? Then I can finish this thing up for good :)